Some Improvements

karaktaka opened the pull request:
Hey,

I found your dashboard and wanted to use it with your script to import my netatmo data into grafana, but it lacked some functionality I needed, like influxdb username and password so I added this along with config file and environment variables support and also a docker container.

Currently the docker container is hosted in my own account at docker hub. Feel free to use it or change it to anything else.

Hope you find this usefull.

Commit:
Docker and Configs

* Added Docker Container
* Added Config file support
* Added Environemnt Variable support to work with Docker easily
* Added docker-compose Files

Commit:
Git in Docker

Need git in docker container

Commit:
Update docker-compose-dev.yml

renamed file, because of docker autobuild

Commit:
Update netatmo_influx.py

Deprecated function

karaktaka commented:
Oh, I almost forgot. I saw that User @dkozinn added support for wind and rain modules.
I also edited your dashboard for my own needs and added additional rows for these modules. If you want I can also share the json.

dkozinn commented:
@karaktaka  I'd be curious to see a screenshot of your dashboard. I modified things significantly enough so that it's not longer quite as "generic" and able to work with multiple stations as well. It might be possible to eventually modify that but I've found that I just haven't had enough time to "generalize" things. 

I like what you did with the configuration file. In my case, I've got it running on the same server as Influx but I don't like hard-coding the credentials.

One other thing that I did was to write a program to calculate Heat Index, Wind Chill, and Dewpoint. I'm am still actively tinkering with it and I'm not particularly proud of the code, but if you're interested it's hi.py (hi=Heat Index) in my repo https://github.com/dkozinn/weather-station. Note that one of the issues is that I built this in pieces and in some cases the values are stored in degrees C and in other cases degrees F. At some point I'll make it consistent and update the database but I figured I'd share it in the current state if you're interested.

mnin commented:
What is missing here:

- we need to install the `dcron` package too
- we need `tini` (installing it and using it to start `crond` in foreground mode via `entrypoint.sh`)
- install the required packages via pip WITH `trusted-host` like `RUN pip install --trusted-host pypi.python.org -r requirements.txt` in the `Dockerfile`
- please adapt my tiny but important fix in #6 to fetch data for each station, not just simulate stations support and put the same values for all stations into the influxdb

Commit:
SSL

Add SSL for Influx

Commit:
Fix SSL

Set use_ssl correctly

Commit:
Merge branch 'master' of https://github.com/florianbeer/grafana-netatmo into florianbeer-master

karaktaka commented:
> 
> 
> What is missing here:
> 
>     * we need to install the `dcron` package too
Why do we need dcron? What's wrong with crond?

>     * we need `tini` (installing it and using it to start `crond` in foreground mode via `entrypoint.sh`)
Do you think zombies will be that much of a problem? I use my image for several days now and didn't ran into any problems so far.

mnin commented:
Thanks!

At the end I solved it easily with just an infinite loop and an sleep:

```shell
while true
do
  /netatmo.py
  sleep 300
done
```

I had couple of issues with the derivates of `crond`, that's why I recommended to use all the time at the and tini and dcron on alpine.

mnin commented:
When I solved the same problem months ago to fetch data from my Fritz!Box and send it to an InfluxDB as well, I decided then to solve it with Python:

```python
import time
import signal

running = True

...

def shutdown(signal, frame):
    global running
    running = False

def main():
    signal.signal(signal.SIGTERM, shutdown)
    signal.signal(signal.SIGINT, shutdown)

    while running:
        ...
        time.sleep(60)

if __name__ == "__main__":
    main()
```

karaktaka commented:
@dkozinn I tried to catch everything without the need to make several screenshots.
![grafik](https://user-images.githubusercontent.com/1451140/90476672-445c5800-e12a-11ea-9cd7-a6c2ab1056f3.png)
I'm not entirely happy with it, as it still shows the battery and signal strength for the base module instead of the wifi signal.

I also thought about adding a heat index or maybe a forecast like the one on the netatmo interface.

Commit:
Update requirements.txt

Use new version in pypi instead of fetching it directly from git

Commit:
Use tini and ditched cron

* Added tini as new init process and ditched cron as suggested by user mnin
* also use his suggestion of running python directly in a loop

karaktaka commented:
@mnin I added tini, ditched cron completely and added your while loop suggestion. Seems like the better solution.

I also added the possibility to connect to InfluxDB with SSL a few commits earlier.
And added a customizable interval setting for querying the netatmo api.

Commit:
Fix

Set execute bit for netatmo_influx.py
Added interval to readme

Commit:
Fix#2

Seems like it is not possible to set the execute bit on windows...

mnin commented on netatmo_influx.py:
```diff
@@ -84,65 +96,75 @@ def parse_args():
+        interval = int(environ.get("INTERVAL"))
+
+    while running:
+        signal.signal(signal.SIGTERM, shutdown)
```
We don't have to do the linking with the signals several times, it is sufficient to do it when starting the main function.

mnin commented:
@karaktaka cool, we don't need tini then ;-)

Commit:
removed tini

* Moved signal outside of while loop
* removed tini

mnin commented on netatmo_influx.py:
```diff
@@ -47,6 +47,9 @@ def shutdown(_signal):
     config = parse_config(args.config)
 
+    signal.signal(signal.SIGTERM, shutdown)
+    signal.signal(signal.SIGINT, shutdown)
```
👍

mnin commented:
I would merge it :D

timmillwood commented:
This PR is awesome! 🥰

karaktaka committed:
Multi-Arch Build

added build hook for docker hub

mnin commented on hooks/build:
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+docker buildx build --platform linux/amd64,linux/arm64,linux/arm/v7 -f $DOCKERFILE_PATH -t $IMAGE_NAME .
```
@karaktaka thanks for sharing!

karaktaka replied:
That's currently not working if run on docker hub. I don't know exactly why, but I can build it like this locally.

karaktaka committed:
Update build

use new builder

karaktaka committed:
builder

* push image
* fix old arm architecture python foo

karaktaka committed:
Update build

karaktaka committed:
Workflow

Use github workflow instead of docker hub

karaktaka committed:
Update build.yml

karaktaka committed:
Merge pull request #2 from karaktaka/test

Github Workflow to build Docker Images

karaktaka committed:
Rewrite

* use pyatmo as netatmo client
* use influxdb as influxdb client
* updated to be used with influxdb 2.0

karaktaka committed:
Update README.md

karaktaka committed:
Update Dockerfile

* add build-base

karaktaka committed:
Update Dockerfile

typo

karaktaka committed:
rewrite

* added linter tests
* added config template

karaktaka committed:
FluxQL

* Added Dashboard with FluxQL
* updated Readme

karaktaka committed:
Merge pull request #3 from karaktaka/rewrite

Rewrite

karaktaka commented:
I'll just leave this here as it seems this repository is dead. I updated my fork to InfluxDB 2.0.
As i don't think this will ever be merged, you might as well just use my fork.
